selfdrive/locationd/test/test_glonass_kaitai.cc

```diff
@@ -0,0 +1,368 @@
+
+  kaitai::kstream str4(inp_data);
+  glonass_t str4_data(&str4);
+  glonass_t::string_4_t* s4 = static_cast<glonass_t::string_4_t*>(str4_data.data());
+
+  int mul = s4->tau_n_sign() ? (-1) : 1;
+  REQUIRE(s4->tau_n() == (data[ST4_TAU_N_V_IDX].second * mul));
+  mul = s4->delta_tau_n_sign() ? (-1) : 1;
+  REQUIRE(s4->delta_tau_n() == (data[ST4_DELTA_TAU_N_V_IDX].second * mul));
+  REQUIRE(s4->e_n() == data[ST4_E_N_IDX].second);
+  REQUIRE(s4->not_used_1() == data[ST4_NU_1_IDX].second);
+  REQUIRE(s4->p4() == data[ST4_P4_IDX].second);
+  REQUIRE(s4->f_t() == data[ST4_F_T_IDX].second);
+  REQUIRE(s4->not_used_2() == data[ST4_NU_2_IDX].second);
+  REQUIRE(s4->n_t() == data[ST4_N_T_IDX].second);
+  REQUIRE(s4->n() == data[ST4_N_IDX].second);
+  REQUIRE(s4->m() == data[ST4_M_IDX].second);
+}
+
+TEST_CASE("parse_string_number_5"){
+  string_data data = generate_string_data(5);
+  string inp_data = generate_inp_data(data);
+
+  kaitai::kstream stream(inp_data);
+  glonass_t gl_string(&stream);
+
+  REQUIRE(gl_string.idle_chip() == data[IDLE_CHIP_IDX].second);
+  REQUIRE(gl_string.string_number() == data[STRING_NUMBER_IDX].second);
+  REQUIRE(gl_string.hamming_code() == data[ST5_HC_OFF + HC_IDX].second);
+  REQUIRE(gl_string.pad_1() == data[ST5_HC_OFF + PAD1_IDX].second);
+  REQUIRE(gl_string.superframe_number() == data[ST5_HC_OFF + SUPERFRAME_IDX].second);
+  REQUIRE(gl_string.pad_2() == data[ST5_HC_OFF + PAD2_IDX].second);
+  REQUIRE(gl_string.frame_number() == data[ST5_HC_OFF + FRAME_IDX].second);
+
+  kaitai::kstream str5(inp_data);
+  glonass_t str5_data(&str5);
+  glonass_t::string_5_t* s5 = static_cast<glonass_t::string_5_t*>(str5_data.data());
+
+  REQUIRE(s5->n_a() == data[ST5_N_A_IDX].second);
+  REQUIRE(s5->tau_c() == data[ST5_TAU_C_IDX].second);
+  REQUIRE(s5->not_used() == data[ST5_NU_IDX].second);
+  REQUIRE(s5->n_4() == data[ST5_N_4_IDX].second);
+  REQUIRE(s5->tau_gps() == data[ST5_TAU_GPS_IDX].second);
+  REQUIRE(s5->l_n() == data[ST5_L_N_IDX].second);
+}
+
+TEST_CASE("parse_string_number_NI"){
+  string_data data = generate_string_data((rand() % 10) +  6);
+  string inp_data = generate_inp_data(data);
+
+  kaitai::kstream stream(inp_data);
+  glonass_t gl_string(&stream);
+
+  REQUIRE(gl_string.idle_chip() == data[IDLE_CHIP_IDX].second);
+  REQUIRE(gl_string.string_number() == data[STRING_NUMBER_IDX].second);
+  REQUIRE(gl_string.hamming_code() == data[ST6_HC_OFF + HC_IDX].second);
+  REQUIRE(gl_string.pad_1() == data[ST6_HC_OFF + PAD1_IDX].second);
+  REQUIRE(gl_string.superframe_number() == data[ST6_HC_OFF + SUPERFRAME_IDX].second);
+  REQUIRE(gl_string.pad_2() == data[ST6_HC_OFF + PAD2_IDX].second);
+  REQUIRE(gl_string.frame_number() == data[ST6_HC_OFF + FRAME_IDX].second);
+
+  kaitai::kstream strni(inp_data);
+  glonass_t strni_data(&strni);
+  glonass_t::string_non_immediate_t* sni =
+    static_cast<glonass_t::string_non_immediate_t*>(strni_data.data());
```

Comment:
why are these on different lines? we don't enforce an 80 char line max

Reply:
its just cause I prefer 80chars len